Exception.js

```diff
@@ -0,0 +1,34 @@
+// @flow
+import type { Eff } from './Eff'
+import { inj, runEff, map } from './Eff'
+import type { Either } from './Either'
+import { left, right } from './Either'
+
+// This effect is used to annotate code which possibly throws exceptions
+export class EXCEPTION {}
+
+export function error(message: string): Error {
+  return new Error(message)
+}
+
+// Throw an exception
+export function throwException<A>(e: Error): Eff<{ err: EXCEPTION }, A> {
+  return inj(() => { throw e })
+}
+
+// Catch an exception by providing an exception handler
+export function catchException<E: Object, A>(handler: (e: Error) => Eff<E, A>, eff: Eff<E & { err: EXCEPTION }, A>): Eff<E, A> {
+  return inj(() => {
+    try {
+      return runEff(eff)
+    } catch (e) {
+      return runEff(handler(e))
+    }
+  })
+}
+
+// Runs an Eff and returns eventual Exceptions as a `Left` value. If the
+// computation succeeds the result gets wrapped in a `Right`.
+export function tryEff<E: Object, A>(eff: Eff<E & { err: EXCEPTION }, A>): Eff<E, Either<Error, A>> {
+  return catchException((e) => inj(() => left(e)), map(right, eff))
+}
```

Comment:
[`\n`][1] ;)


[1]: https://github.com/editorconfig/editorconfig/wiki/Newline-at-End-of-File-Support